tensorboard/webapp/widgets/dropdown/dropdown_test.ts

```diff
@@ -0,0 +1,150 @@
+/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {Component, Input} from '@angular/core';
+import {ComponentFixture, TestBed} from '@angular/core/testing';
+import {MatSelectModule} from '@angular/material/select';
+import {By} from '@angular/platform-browser';
+import {DropdownComponent, DropdownOption} from './dropdown_component';
+
+/**
+ * Test class that uses the <tb-dropdown> component.
+ */
+@Component({
+  selector: 'testing-component',
+  template: `
+    <tb-dropdown
+      [value]="expId"
+      [options]="configOptions"
+      (selectionChange)="expChanged($event)"
+    ></tb-dropdown>
+  `,
+})
+class TestableComponent {
+  @Input() expId!: string;
+  @Input() configOptions!: DropdownOption[];
+
+  expChanged(value: string) {
```

Comment:
I saw the implementation is used in link 148 but what we want to do is have a spy and check if the spy is called with desired param. To be more specific, you can replace this with `@Input() selectionChange!: (value: string) => void`

later in line 62. you create a spy and assign to this input

```
selectionChangeSpy = jasmine.createSpy();
...
fixture.componentInstance.selectionChange = selectionChangeSpy;
```

then in tests we do 
```
expect(selectionChangeSpy).toHaveBeenCalledWith('2')
```

you can check the example `onSingleValueChanged` here
https://cs.opensource.google/tensorflow/tensorboard/+/master:tensorboard/webapp/widgets/range_input/range_input_test.ts;l=66?q=range_input&ss=tensorflow%2Ftensorboard

Reply:
Sorry this comment is cached and sent again. Please ignore it